rsopt/configuration/options/__init__.py

```diff
@@ -69,6 +68,21 @@ def _check_options(cls, options):
         for key in cls.REQUIRED_OPTIONS:
             assert options.get(key), f"{key} must be defined in options for {name}"
 
+    @property
+    def objective_function(self) -> typing.List[str]:
+        _obj_f = self._objective_function.copy()
+        if len(self._objective_function) == 2:
```

Comment:
This seems risky on two accounts

1. What is self._objective_function is still an empty list?
2. It is a list of strings, so you're assuming a length 2, but not more, is the first part as a path

I see the assert in the setter about this, but what does a `len(of) == 3` do then? It might be better to pass in "path" as a mapped argument instead to the setter, and if it is None, then do whatever your normal processing is.

The input is basically unsanitized and out of the control of the code, so I suppose it depends what needs to be validated